compiler-core/src/type_.rs

```diff
@@ -46,25 +46,71 @@ pub trait HasType {
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum Type {
+    /// App is the primary "wrapper" for most types. The type may come from a
+    /// third party module, but if `module` is empty that means the type is
+    /// coming from the gleam prelude instead (where Int, Float, String, etc is
+    /// defined). If the `App` takes generic arguments they will be put in the
+    /// `args` field.
+    ///
+    /// # Example
+    ///
+    /// ```gleam
+    /// Int //=> Type::App { name: "Int", args: vec![], module: vec![], public: true }
+    /// String //=> Type::App { name: "String", args: vec![], module: vec![], public: true }
+    /// ```
```

Comment:
```suggestion
    /// A nominal (named) type such as `Int`, `Float`, or a programmer defined
    /// custom type such as `Person`. The type can take other types as
    /// arguments (aka "generics" or "parametric polymorphism").
    ///
    /// If the type is defined in the Gleam prelude the `module` field will be
    /// empty, otherwise it will contain the name of the module that
    /// defines the type.
```